src/d3d9/d3d9_swapchain.cpp

```diff
@@ -9,6 +9,25 @@
 
 namespace dxvk {
 
+  static std::unordered_map<HWND, WNDPROC> windowProcMap;
+
+
+  static LRESULT CALLBACK WindowProcA(HWND window, UINT message, WPARAM wparam, LPARAM lparam)
+  {
+      if (message == WM_NCCALCSIZE && wparam == TRUE)
+        return 0;
+      return CallWindowProcA(windowProcMap[window], window, message, wparam, lparam);
+  }
+
+
+  static LRESULT CALLBACK WindowProcW(HWND window, UINT message, WPARAM wparam, LPARAM lparam)
+  {
+      if (message == WM_NCCALCSIZE && wparam == TRUE)
+        return 0;
+      return CallWindowProcW(windowProcMap[window], window, message, wparam, lparam);
+  }
```

Comment:
Can we merge this? This needs some form of mutex anyway, so I am sure we can shove in some extra data long with WNDPROC to make this

```suggestion
  static LRESULT CALLBACK WindowProc(HWND window, UINT message, WPARAM wparam, LPARAM lparam) {
    auto windowData = windowMap[window];
    if (message == WM_NCCALCSIZE && wparam == TRUE)
      return 0;
    return windowData.unicode
      ? CallWindowProcW(windowData.proc, window, message, wparam, lparam)
      : CallWindowProcA(windowData.proc, window, message, wparam, lparam);
  }
```